cmd/argocd/commands/utils/claims.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+// ArgoClaims defines the claims structure based on Dex's documented claims
+type ArgoClaims struct {
+	jwt.RegisteredClaims
+	Email         string   `json:"email,omitempty"`
+	EmailVerified bool     `json:"email_verified,omitempty"`
+	Name          string   `json:"name,omitempty"`
+	Groups        []string `json:"groups,omitempty"`
+	// As per Dex docs, federated_claims has a specific structure
+	FederatedClaims *FederatedClaims `json:"federated_claims,omitempty"`
+}
+
+// FederatedClaims represents the structure documented by Dex
+type FederatedClaims struct {
+	ConnectorID string `json:"connector_id"`
+	UserID      string `json:"user_id"`
+}
+
+// GetFederatedClaims extracts federated claims from jwt.MapClaims
+func GetFederatedClaims(claims jwt.MapClaims) *FederatedClaims {
```

Comment:
Should create the full ArgoClaims object from MapClaims. The implementation is very similar to 
`func MapClaims(claims jwtgo.Claims) (jwtgo.MapClaims, error)`, except that it unmarshal the ArgoClaims object